libevent-sys/build.rs

```diff
@@ -1,18 +1,105 @@
 use std::env;
 use std::path::PathBuf;
 
-use bindgen;
-use pkg_config;
+#[cfg(feature = "bundled")]
+fn build_libevent(libevent_path: &str) -> PathBuf {
+    let mut config = cmake::Config::new(libevent_path);
 
-fn main() {
-    // Use pkg-config to find and link libevent
-    pkg_config::Config::new()
+    // TODO: Or just both and decide elsewhere?
+    if cfg!(feature = "static") {
+        config.define("EVENT__LIBRARY_TYPE", "STATIC");
+    } else {
+        config.define("EVENT__LIBRARY_TYPE", "SHARED");
+    }
+
+    // Disable unnecessary dev-only features
+    {
+        config.define("EVENT__DISABLE_BENCHMARK", "ON")
+            .define("EVENT__DISABLE_TESTS", "ON")
+            .define("EVENT__DISABLE_REGRESS", "ON")
+            .define("EVENT__DISABLE_SAMPLES", "ON");
+
+        // TODO: Address policy warnings for cross-compilation?
+        /*config.define("CMAKE_POLICY_DEFAULT_CMP0056", "NEW")
+            .define("CMAKE_POLICY_DEFAULT_CMP0066", "NEW")*/
+    }
+
+    let dst = config.build();
+
+    println!("cargo:rustc-link-search={}/lib", dst.display());
+
+    // Library 'event' is considered deprecated, so link each sub-component
+    // individually.
+    println!("cargo:rustc-link-lib=static=event_core");
+    println!("cargo:rustc-link-lib=static=event_extra");
+
+    println!("cargo:include={}/include", dst.display());
+
+    dst
+}
+
+#[cfg(not(feature = "bundled"))]
+fn run_pkg_config() -> Option<Vec<String>> {
+    let mut pkg = pkg_config::Config::new();
+    pkg.cargo_metadata(false)
         .atleast_version("2")
-        .statik(cfg!(feature = "static"))
-        .probe("libevent")
-        .unwrap();
+        .statik(cfg!(feature = "static"));
+
+    let mut lib = match pkg.probe("libevent_core") {
+        Ok(lib) => lib,
+        Err(_e) => { return None; }
+    };
+    // TODO: Probably combine all pc includes, just to be safe.
+
+    pkg.cargo_metadata(true).probe("libevent_core").unwrap();
```

Comment:
This seems to be duplicating https://github.com/jmagnuson/libevent-rs/pull/1/commits/3f83d47b2b13384c6a44f11aa1954480d67b982d#diff-adab382290cb5e55b3ad3eb478c2c84fR48

Reply:
So I lifted this pattern from [curl-sys](https://github.com/alexcrichton/curl-rust/blob/33b7a0b3eba194b5b13a66e38950f6076f0c91c1/curl-sys/build.rs#L457) thinking that it had to do with operation ordering between emitting includes and libs. But looking again, I wonder if it has to do with preventing emission of libs in case `curl_config_reports_http2` fails out a few lines up. But if that's the case, why not just do the check first-thing?